java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -59,6 +62,23 @@ public void visitNode(Tree tree) {
     }
   }
 
+  private static boolean isUtilityClass(ClassTree classTree) {
+    return hasOnlyStaticMembers(classTree) && !anonymousClass(classTree) && !extendsAnotherClassOrImplementsSerializable(classTree)
+      && !containsMainMethod(classTree);
+  }
+
+  private static boolean containsMainMethod(ClassTree classTree) {
+    return classTree.members().stream()
+      .filter(member -> member.is(Tree.Kind.METHOD))
+      .anyMatch(method -> isMainMethod((MethodTree) method));
+  }
+
+  private static boolean isMainMethod(MethodTree method) {
+    return MAIN.matches(method)
+      && method.returnType().symbolType().is("void")
+      && ModifiersUtils.hasModifier(method.modifiers(), Modifier.PUBLIC);
```

Comment:
It is checked before by `hasOnlyStaticMembers` , but having a second thought about it, it will be perhaps more consistent to check for it again. I'll fix that